### DISCUSSION TITLE
v6.0.0

### DISCUSSION THREAD

**Discussion by oojacoboo** [thecodingmachine/graphqlite#484] [Announcements]
Hurrah version 6 landing!  There is a lot in this one and it's long overdue, but we finally hit a great spot, so without further adieu.

Version 6 brings `#[Input]` defined InputTypes way forward and solidifies their overall functionality.  `#[Input]` attribute defined InputTypes are becoming my preferred way of defining these types.  One single DTO or even Doctrine entity (if you so dare!) can be used to define both input and output types, with virtually no side-effects or missing feature functionality.  Prior to version 6, `#[Input]` defined types were rather limited.  This is no longer the case!  If you haven't had a chance yet, now is the time to give them a spin! 👍  A special thanks to @Lappihuan for all his teamwork on the InputType improvements.  This wouldn't have happened without him!

In addition to the huge improvements made on InputTypes, we now have support for native Enums, union return typing, and generics support!

#### Dependencies:
- https://github.com/thecodingmachine/graphqlite/commit/652440a4a0fa77f327446e89fabb30bb044ad3d4 Symfony ^6 is now supported.

#### Breaking Change:
https://github.com/thecodingmachine/graphqlite/pull/458 With `Input` annotated InputTypes, if you're using the same class for multiple types, like an update type and create type (also now a possibility), if you provide a name, you must also provide the `default` parameter if you want the InputType to be the "default".

```php
#[Input] // Is default
#[Input(default: false)] // Not default
#[Input(name: 'SomethingInput', default: true] // Is default
#[Input(name: 'UpdateSomethingInput'] // Not default
```

#### New Features:
https://github.com/thecodingmachine/graphqlite/pull/409 Native Enum support! 🎉 
https://github.com/thecodingmachine/graphqlite/pull/467 Support native union return typing
https://github.com/thecodingmachine/graphqlite/pull/468 Support generic phpdpc style typed interables for output type values
https://github.com/thecodingmachine/graphqlite/pull/435 New Custom InputType Validator offering flexibility in how you can handle validation on all of your `#[Input]` attributes.
https://github.com/thecodingmachine/graphqlite/pull/458 Recursive InputType resolving
https://github.com/thecodingmachine/graphqlite/pull/466 Bring existing getter `#[Field]` defined functionality from output `#[Type]` objects to InputType annotated fields, via setters.

<hr /><em>This discussion was created from the release <a href='https://github.com/thecodingmachine/graphqlite/releases/tag/v6.0.0'>v6.0.0</a>.</em>

**Comment by cvergne** [comment]
👋 Hi,
Sorry it's not about this repo, but a directly related one.
Is the bundle still maintained ?
[Many of us are waiting for a new bundle release](https://github.com/thecodingmachine/graphqlite-bundle/issues/142) since all dependencies updates, for compatibility with Symfony 6.1+.

At least if we can have information if something is blocking, just to know how to handle it :)

Thanks for your awesome work, and your time on this subject 🙏

**Reply by nguyenk** [reply]
@cvergne I just drafted realease [v6.0.0-beta.0](https://github.com/thecodingmachine/graphqlite-bundle/tree/v6.0.0-beta.0)

As I'm not using the bundle in any project right now, I'd need you to test. I just made is possible to instal, and fixed the CI, so please report, or event better submit a PR for fixes.
